connection_test.go

```diff
@@ -139,6 +139,53 @@ func TestGetQueryResultUsesTokenFromTokenAccessor(t *testing.T) {
 	}
 }
 
+func TestGetQueryResultTokenExpiry(t *testing.T) {
```

Comment:
It's also worth having another test for custom token accessor?

Reply:
Tests for simpleTokenAccessor with and without token set should be enough.